cni/pkg/plugin/cnieventclient.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	cniv1 "github.com/containernetworking/cni/pkg/types/100"
+
+	"istio.io/istio/cni/pkg/nodeagent"
+)
+
+// newCNIClient is a unit test override variable for mocking.
+var newCNIClient = buildClient
+
+// An udsCore write entries to an UDS server with HTTP Post. Log messages will be encoded into a JSON array.
+type CNIEventClient struct {
+	client *http.Client
+	url    string
+}
+
+func buildClient(address, path string) CNIEventClient {
+	c := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", address)
+			},
+		},
+		Timeout: 1000 * time.Millisecond,
+	}
+	eventC := CNIEventClient{
+		client: c,
+		url:    "http://unix" + path,
```

Comment:
what is the  `unix` here,  place holder?

Reply:
It's a quirk of the Go httpclient - it doesn't like/parse `unix://`, and AFAICT this is the Go convention for http-over-uds URLs.

Reply:
ACK Learned that yesterday